File: 04.Simulation模拟/leetcode_3210_numberOfSpecialChars.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int numberOfSpecialChars(string word) {
        unordered_map<char, int> cnt;
        for (auto& c : word) cnt[c]++;
        int res = 0;
        for (int i=0; i<26; i++){
            if (cnt['a'+i] && cnt['A'+i]){
                res++;
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    string word = "aaAbcBC";
    cout << s.numberOfSpecialChars(word) << endl;
}

File: 04.Simulation模拟/leetcode_3210_numberOfSpecialChars.py
from collections import Counter

class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        cnt = Counter(word)
        return sum(1 for i in range(26) if cnt[chr(ord('a')+i)] and cnt[chr(ord('A')+i)])

if __name__ == "__main__":
    s = Solution()
    word = "aaAbcBC"
    print(s.numberOfSpecialChars(word))
